test/metric_learn_test.py

```diff
@@ -1136,6 +1136,19 @@ def test_changed_behaviour_warning_random_state(self):
       rca_supervised.fit(X, y)
     assert any(msg == str(wrn.message) for wrn in raised_warning)
 
+  def test_unknown_labels(self):
+    n = 100
+    X, y = make_classification(random_state=42, n_samples=2 * n)
+    y2 = np.concatenate((y[:n], np.ones(n)))
+
+    rca = RCA_Supervised(num_chunks=2)
+    rca.fit(X[:n], y[:n])
+
+    rca2 = RCA_Supervised(num_chunks=2)
+    rca2.fit(X, y2)
+
+    np.testing.assert_array_equal(rca.components_, rca2.components_)
+
```

Comment:
There seems to be some problems with this test:
- Currently there is no unlabeled points. On line 1142 I think you want `-np.ones(n)` and not `np.ones(n)`
- The reason why the test passes nevertheless is because RCA raises a warning "The inner covariance matrix is not invertible" and returns `rca.components_` with all nans in both cases. As suggested by the warning you may need to  first reduce the dimension using PCA

Reply:
Done